public/app/plugins/datasource/prometheus/dataquery.cue

```diff
@@ -0,0 +1,48 @@
+// Copyright 2023 Grafana Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grafanaplugin
+
+import (
+	common "github.com/grafana/grafana/packages/grafana-schema/src/common"
+	"github.com/grafana/grafana/pkg/plugins/pfs"
+)
+
+// This file (with its sibling .cue files) implements pfs.GrafanaPlugin
+pfs.GrafanaPlugin
+
+composableKinds: DataQuery: {
+	maturity: "merged"
+
+	lineage: {
+		seqs: [
+			{
+				schemas: [
+					{
+						common.DataQuery
+						expr:        string
+						instant?:    bool
+						range?:      bool
```

Comment:
Can both range and instant be true?

Reply:
they can't, and we can natively constrain them in CUE to indicate that invariant, but at the moment cuetsy evaluates the operation and converts the TS output to just `false` - see grafana/cuetsy#75

Reply:
Yes, it can be. In query options there is a `type` option field:
![image](https://user-images.githubusercontent.com/820946/223843061-acd137eb-0f2c-4d08-b4c7-04b192a83323.png)

Reply:
oh! i totally misunderstood this on my first pass through it, then :)